Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -90,16 +231,79 @@ pathManagerState& cruisingState::getInstance()
     return singleton;
 }
 
+
 void coordinateTurnElevation::execute(pathManager* pathMgr)
 {
+    
+    SFOutput_t * sensFusionOutput = sensorFusion::GetSFOutput(); // Get sensor fusion data
+    IMU_Data_t * imudata = sensorFusion::GetIMUData(); // Gets raw IMU data
+    _WaypointManager_Data_Out * waypointOutput = cruisingState::GetOutputData(); // Get output data from waypoint manager
+    
+    CoordinatedTurnInput_t turnInput {};
+    AltitudeAirspeedInput_t altAirspeedInput {};                                   
+
+    
     //get elevation and turning data
-    if(isError)
+    //loading in commands data from each of the states
+    switch(pathMgr->stage){
+        case ROLL:
+            LandingManager::translateLTSFCommandsToCoordTurns(takeoffRollStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case CLIMB:
+            LandingManager::translateLTSFCommandsToCoordTurns(takeoffClimbStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case TRANSITION:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingTransitionStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case SLOPE:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingSlopeStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case FLARE:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingFlareStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case DECRAB:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingDecrabStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
```

Comment:
I think the only thing changing here is where your'e getting the control input from. So that should be the only thing in the switch statement, the common stuff should just be a single line below the switch statement.

Reply:
the cruising state outputs are accessed completely differently though, that's why I just have the accessing functions in each case statement. I'm not sure how else I can implement this more simply.

Reply:
What I mean is, you can make it so that rather than having all this in each statement

 LandingManager::translateLTSFCommandsToCoordTurns(landingDecrabStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
 
 you can have:
 
 ControlInput_t controlInput =  LandingDecrabStage::getControlOutput();
 
 (with the variabe being asigned whatever it needs to be, then at the bottom, you can have a single call to the function with:
 
  LandingManager::translateLTSFCommandsToCoordTurns(controlInput , sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
  
  
 The 2 reasons I note this:
  
  I'm sure you've heard it said that repetition in software is bad. This is a case of that, if for whatever reason, you have to change the names of the other args or of the function or whatever, you'l have to make that change in each case with your current implementation, which is prone to mistakes and is a bit of a pain.
  
  Second reason is that this current implementation forces readers to read every line, playing "find the differences" to figure out what your'e trying to do. That's much more tough than being able to just scan it over.
  
  
  
  
  Is it the biggest issue in the world ? no. But if you get in the habbit of doing little things like this, it'l slowly make you a better developer.

Reply:
ok gotcha, I'll make those changes